File: Classes/Premade/MASPreferences/UpdatePreferenceViewController.m
//
//  UpdatePreferenceController.m
//  BeFit
//
//  Created by Jon Brown on 1/2/14.
//
//

#import "UpdatePreferenceViewController.h"

@implementation UpdatePreferenceViewController

- (id)init
{
    return [super initWithNibName:@"UpdatePreferencesView" bundle:nil];
}

#pragma mark -
#pragma mark MASPreferencesViewController

- (NSString *)viewIdentifier
{
    return @"UpdatePreferences";
}

- (NSImage *)toolbarItemImage
{
    return [NSImage imageNamed:@"Update"];
}

- (NSString *)toolbarItemLabel
{
    return NSLocalizedString(@"Update", @"Toolbar item name for the Update preference pane");
}

- (void) awakeFromNib
{
    
    
    NSColor *color = [NSColor redColor];
    NSMutableAttributedString *colorTitle =
    [[NSMutableAttributedString alloc] initWithAttributedString:[button attributedTitle]];
    
    NSRange titleRange = NSMakeRange(0, [colorTitle length]);
    
    [colorTitle addAttribute:NSForegroundColorAttributeName
                       value:color
                       range:titleRange];
    
    [button setAttributedTitle:colorTitle];

}

@end

File: Classes/Premade/MASPreferences/UpdatePreferenceViewController.swift
//
//  UpdatePreferenceViewController.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/8/19.
//

import Foundation
import Cocoa
import MASPreferences

class UpdatePreferenceViewController: NSViewController, MASPreferencesViewController {
    @IBOutlet var button: NSButton!
    
    init() {
        super.init(nibName: "UpdatePreferencesView", bundle: nil)
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    //MARK: - MASPreferencesViewController
    
    var viewIdentifier: String {
        return "UpdatePreferences"
    }
    
    var toolbarItemLabel: String? {
        return NSLocalizedString("Update", comment: "Toolbar item name for the Update preference pane")
    }
    
    var toolbarItemImage: NSImage? {
        return NSImage(named: "Update")
    }
    
    //MARK: - Methods
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        let color = NSColor.red
        let colorTitle = NSMutableAttributedString(attributedString: button.attributedTitle)
        let titleRange = NSRange(location: 0, length: colorTitle.length)
        
        colorTitle.addAttribute(.foregroundColor, value: color, range: titleRange)
        
        button.attributedTitle = colorTitle
    }
}
